pkg/controller/snapshot_controller.go

```diff
@@ -824,13 +825,19 @@ func (ctrl *csiSnapshotController) SetDefaultSnapshotClass(snapshot *crdv1.Volum
 
 // getClaimFromVolumeSnapshot is a helper function to get PVC from VolumeSnapshot.
 func (ctrl *csiSnapshotController) getClaimFromVolumeSnapshot(snapshot *crdv1.VolumeSnapshot) (*v1.PersistentVolumeClaim, error) {
-	if snapshot.Spec.Source == nil || snapshot.Spec.Source.Kind != pvcKind {
-		return nil, fmt.Errorf("The snapshot source is not the right type. Expected %s, Got %v", pvcKind, snapshot.Spec.Source)
+	if snapshot.Spec.Source == nil {
+		return nil, fmt.Errorf("the snapshot source is not specified.")
+	}
+	if snapshot.Spec.Source.Kind != pvcKind {
+		return nil, fmt.Errorf("the snapshot source is not the right type. Expected %s, Got %v", pvcKind, snapshot.Spec.Source.Kind)
 	}
 	pvcName := snapshot.Spec.Source.Name
 	if pvcName == "" {
 		return nil, fmt.Errorf("the PVC name is not specified in snapshot %s", snapshotKey(snapshot))
 	}
+	if snapshot.Spec.Source.APIGroup != nil && *(snapshot.Spec.Source.APIGroup) != apiGroup {
+		return nil, fmt.Errorf("the snapshot source does not have the right APIGroup. Expected %s, Got %s", apiGroup, *(snapshot.Spec.Source.APIGroup))
+	}
```

Comment:
It will print out like "Expected , Got xxx", I wonder it may be better to print out like
Expected "", Got "xxx"? or Expected empty string, Got "xxx"?

Reply:
Sure.